### ISSUE TITLE
D. Modeling

### ISSUE THREAD

**Issue by edaaydinea** [edaaydinea/-#6]
- [ ] D.1- Data Transformation and Encoding
  - [ ] D.1.1- Transformation (Scaling Numerical Variables)
  - [ ] D.1.2- Encoding Categorical Variables (Classic/Contrast/Bayesian)
- [ ] D.2. Metric, Baseline and Estimator/Classifier Selection
  - [ ] D.2.1- Metric and Baseline Selection for Model Evaluation
    - [ ] D.2.1.1- Metric Selection
    - [ ] D.2.1.2- Baseline Selection
- [ ] D.3- Estimator/Classifier Selection
  - [ ] D.3.1- K-Nearest Neighbors (KNN)
  - [ ] D.3.2- Naive Bayes Classifier
  - [ ] D.3.3- Logistic Regression Classifier
  - [ ] D.3.4- Support Vector Machines Classifier
  - [ ] D.3.5- Neural Network Classifiers
    - [ ] D.3.5.1- Multi-Layer Perceptron (MLP)
    - [ ] D.3.5.2- Convolutional Neural Networks (CNN)
    - [ ] D.3.5.3- Recurrent Neural Networks (RNN)
  - [ ] D.3.6- Ensembled Classifiers
    - [ ] D.3.6.1- Random Forest Classifier (Bagging)
    - [ ] D.3.6.2- Gradient Boosting Classifier
    - [ ] D.3.6.3- Stacked Generalization (Stacking)
- [ ] D.4- Model Selection (Hyperparameter Tuning)
